hera_mc/geo_handling.py

```diff
@@ -313,33 +322,292 @@ def get_location(self, to_find, query_date, show_location=False, verbosity='m'):
                     print(L, ' not found.')
         return found_location
 
-    def get_all_everconnected_locations(self):
-        """
-        Returns a list of all of the locations ever connected.
+    def _search_loop(self, stn, at_date, now, m, base_tdelta, full_req):
```

Comment:
I really do not feel good about this function. What is its purpose? Where do the various hardcoded magic time intervals come from? Why are `m` and `base_tdelta` separate arguments if they're only ever used in conjunction? I would like someone to double-check its internal logic and add comments documenting how this function works.